website/blog/2022-03-04-modifying-blocks.md

```diff
@@ -0,0 +1,275 @@
+
+```js
+import React, { useMemo } from 'react';
+import classnames from 'classnames';
+import { checkAttr, props, selector, outputCssVariables, getUnique } from '@eightshift/frontend-libs/scripts';
+import { HeadingEditor } from '../../heading/components/heading-editor';
+import { ParagraphEditor } from '../../paragraph/components/paragraph-editor';
+import manifest from './../manifest.json';
+import globalManifest from './../../../manifest.json';
+
+export const QuoteEditor = (attributes) => {
+	const unique = useMemo(() => getUnique(), []);
+	//...
+```
+Next, we need to add a unique `data-id` and output the `<style>` tag containing CSS variables.
+
+Modify your return statement so it looks like this:
+
+```jsx
+return (
+	<>
+		<figure className={quoteClass} data-id={unique}>
+			{outputCssVariables(attributes, manifest, unique, globalManifest)}
+			//...
+```
+
+If you try adding a Quote block in the editor, you should notice that now it has a light blue background with a quote icon in a darker shade of blue. This equals our default value for the `quoteColorTheme` attribute we added in the component's manifest.
+
+### Outputting CSS variables in PHP template
+The PHP template for the Quote component is located in **_src/Blocks/components/quote/quote.php_**. This process is similar to the one described above, it's just written in PHP. All helper methods we need for this are contained inside the `Components` class, which is already included in this file.
+
+Similar to the JS template, we need a unique value, assign it to `data-id`, and output CSS variables. After adding these, the code should look like this:
+
+```php
+<?php
+//...
+$quoteClass = Components::classnames([
+	Components::selector($componentClass, $componentClass),
+	Components::selector($blockClass, $blockClass, $selectorClass),
+	Components::selector($additionalClass, $additionalClass),
+]);
+
+$unique = Components::getUnique();
+?>
+
+<figure class="<?php echo \esc_attr($quoteClass); ?>" data-id="<?php echo \esc_attr($unique); ?>">
+	<?php echo Components::outputCssVariables($attributes, $manifest, $unique, $globalManifest); ?>
+	//...
+```
+
+> Both PHP and JS have the same helpers to make writing code for editor and front view as identical as possible.
+
+### Adding options
+After adding these snippets, you should already see that your block is using the blue color theme, which we defined as the default value for `quoteColorTheme` in the manifest. We want to have an easy way to change the color theme in the editor. Now we'll add a new option which will do just that.
+
+We will add these options in **_src/Blocks/components/quote/components/quote-options.js_**. Again, we will start with the imports we will need. First one is `__` from **_@wordpress/i18n_** package. Additional imports we need are from **_@eightshift/frontend-libs/scripts_**, so we will just add those to the list.
+```js
+import { __ } from '@wordpress/i18n';
+import {
+	//...
```

Comment:
Why ... here?

Reply:
I'm using this to reduce the amount of code in example. Instead of listing everything inside that import, i just added `//...`

Reply:
The imports are present in the previous posts, right? Is so, then it's ok to shorten them 👍🏼

Reply:
The idea behind these code examples isn't to just blindly copy/paste them, but to see which parts are newly added and where with reference to existing code. Since there are quite a bit of imports in this example, I didn't want to list all of them here. I may add an additional explanation somewhere in the post about what `//...` comment means so people aren't confused.

But this is great that you commented on this. I thought this was self-explanatory. Guess I thought wrong 😄